core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -32,7 +32,9 @@ private[v1] class ApplicationListResource(uiRoot: UIRoot) {
       @DefaultValue("3000-01-01") @QueryParam("maxDate") maxDate: SimpleDateParam,
       @QueryParam("limit") limit: Integer)
   : Iterator[ApplicationInfo] = {
-    val allApps = uiRoot.getApplicationInfoList
+    val numApps: Int = if (limit != null) limit else Integer.MAX_VALUE
```

Comment:
`Option(limit).getOrElse(Integer.MaxValue)` is probably a little more idiomatic. At least, using the Scala constant is.